src/main/java/com/gfl/client/security/WorkerApiKeyValidationFilter.java

```diff
@@ -19,10 +21,12 @@
 import java.util.List;
 
 @Component
+@RequiredArgsConstructor
 public class WorkerApiKeyValidationFilter extends OncePerRequestFilter {
 
     @Value("${executor.service.auth.token.value}")
     private String workerApiKey;
+    private RsaManager rsaManager;
```

Comment:
add `final` to prevent NPE:
```suggestion
    private final RsaManager rsaManager;
```